onebusaway-android/src/main/java/org/onebusaway/android/ui/TripDetailsListFragment.java

```diff
@@ -228,19 +234,8 @@ private void setTripDetails(ObaTripDetailsResponse data) {
             getListView().setDivider(null);
             setListAdapter(mAdapter);
 
-            // Scroll to stop if we have the stopId available
-            if (mStopId != null) {
-                mStopIndex = findIndexForStop(mTripInfo.getSchedule().getStopTimes(), mStopId);
-                if (mStopIndex != null) {
-                    listView.post(new Runnable() {
-                        @Override
-                        public void run() {
-                            listView.setSelection(mStopIndex);
-                        }
-                    });
-                }
-            } else {
-                // If we don't have a stop, then scroll to the current position of the bus
+            if (mScrollMode != null && mScrollMode.equals("vehicle")) {
```

Comment:
To simplify this section of code, this can be:

~~~
if (mScrollMode == null) {
    // Don't automatically scroll to stop or vehicle icon
    return;
}
~~~

So, if no scroll mode is provided, we do nothing.  And then do a `switch` for the two scroll modes.  This `switch` should probably be in it's own private method to avoid cluttering this one too much.